user_coffee_shop_query.rb

```diff
@@ -0,0 +1,22 @@
+require_relative 'distance_calculator'
+
+class UserCoffeeShopQuery
+  def initialize(coffee_shops, user)
+    @coffee_shops = coffee_shops
+    @user = user
+  end
+
+  def result
+    distances.sort_by { |coffee_shop| coffee_shop[:distance] }.first(3)
+  end
```

Comment:
As a bonus challenge (only if you like challenges and you have time). Could you optimize this algorithm, IF it can be optimized better, for a huge list of coffee shops. At least more optimized, not really the best solution